plugins/outputs/pulsar/pulsar.go

```diff
@@ -0,0 +1,378 @@
+package pulsar
+
+import (
+	"context"
+	"fmt"
+	"github.com/gofrs/uuid"
+	"log"
+	"strings"
+	"time"
+
+	"github.com/apache/pulsar-client-go/pulsar"
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/plugins/outputs"
+	"github.com/influxdata/telegraf/plugins/serializers"
+)
+
+var ValidTopicSuffixMethods = []string{
```

Comment:
Please make this a private global variable or even better move it to the struct.